registrations/management/commands/revalidate_registrations.py

```diff
@@ -0,0 +1,102 @@
+            "in the SMB service")
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            '--blind', action='store_false', default=True,
+            dest='check_subscription',
+            help=('Do not check with the stage based messaging API whether'
+                  'or not a subscription for the identity already exists.'
+                  'NOT RECOMMENDED AT ALL'))
+        parser.add_argument(
+            '--sbm-url', dest='sbm_url', type=validate_and_return_url,
+            default=environ.get('STAGE_BASED_MESSAGING_URL'),
+            help=('The Stage Based Messaging Service to verify '
+                  'subscriptions for.'))
+        parser.add_argument(
+            '--sbm-token', dest='sbm_token',
+            default=environ.get('STAGE_BASED_MESSAGING_TOKEN'),
+            help=('The Authorization token for the SBM Service')
+        )
+        parser.add_argument(
+            '--invalid-field', dest='invalid_field', default=None,
+            help=("Field that failed to validate when registration was "
+                  "created. Must match 'invalid_field' in registration data")
+        )
+        parser.add_argument(
+            '--batch-size', dest='batch_size', type=int, default=None,
+            help=("The number of registrations to process. This allows batch "
+                  "processing.")
+        )
+
+    def handle(self, *args, **kwargs):
+        sbm_url = kwargs['sbm_url']
+        sbm_token = kwargs['sbm_token']
+        check_subscription = kwargs['check_subscription']
+        invalid_field = kwargs['invalid_field']
+        batch_size = kwargs['batch_size']
+
+        if check_subscription:
+            if not sbm_url:
+                raise CommandError(
+                    'Please make sure either the STAGE_BASED_MESSAGING_URL '
+                    'environment variable or --sbm-url is set.')
+
+            if not sbm_token:
+                raise CommandError(
+                    'Please make sure either the STAGE_BASED_MESSAGING_TOKEN '
+                    'environment variable or --sbm-token is set.')
+            client = StageBasedMessagingApiClient(sbm_token, sbm_url)
+
+        registrations = Registration.objects.filter(
+            validated=False, data__invalid_fields__contains=[invalid_field])
+
+        if batch_size is not None:
+            registrations = registrations[:batch_size]
+
+        count = 0
+        for reg in registrations.iterator():
+            self.log("Validating registration %s" % reg.id)
+            if check_subscription and self.count_subscriptions(client, reg):
+                self.log(('Identity %s already has subscription. Skipping.')
+                         % (reg.registrant_id))
+                continue
+            """
+            validate_subscribe() checks all data for the registration is valid
+            and creates the Subscription Request
+            """
```

Comment:
If I'm not mistaken, docstrings mean something to systems that auto-generate your docs. Otherwise I don't think there's any real difference. But I do think it's convention to use docstrings for things that should be documented (like at the top of methods or tests) and comments in the code so I'll change this. Thanks.